app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -131,6 +139,11 @@ class RecentSyncedTabFeature(
             RecentSyncedTabs.latestSyncedTabIsStale.add()
         }
     }
+
+    companion object {
+        // Number of recent synced tabs we want to keep in the success state.
+        const val MAX_RECENT_SYNCED_TABS = 8
```

Comment:
This seems like a good heuristic to start with. It may be worth capturing metrics around how many tabs are blocked so we know if we run a risk of not displaying anything. I will file a follow up